datalad/interface/__init__.py

```diff
@@ -92,6 +92,7 @@
         ('datalad.distribution.siblings', 'Siblings', 'siblings'),
         ('datalad.support.sshrun', 'SSHRun', 'sshrun'),
         ('datalad.local.subdatasets', 'Subdatasets', 'subdatasets'),
+        ('datalad.local.foreach', 'ForEach', 'foreach'),
```

Comment:
indeed, we could workout `--entity-type={dataset,file,sibling,branch}`! For Python mode might want to establish helper classes to encapsulate those notions or otherwise just be records returned by e.g. `status`, `siblings` (and ?) correspondingly?  
actually -- all of those would be related to `ds` and `refds` anyways, so it might be just a matter of adding extra fields ?

As for command(s) name -- I am still hopeful that we eventually implement and RF to https://github.com/datalad/datalad/issues/4539 , and then it would be `dataset foreach`, `sibling foreach` etc (and I guess get `file` and `branch` top level commands)

Reply:
FWIW: ATM not yet sure if renaming to anything else than `foreach` is worthwhile since `foreach` makes clearer analog to `git submodule foreach` and it is short... moreover, one of the main goals here is to help "crossing datasets boundaries", and it could indeed be then mixed in with also sweeping through various elements within those datasets, so in principle could do what it does + additional loop inside if (optional) `--entity-type` would be given (defaulting to dataset, which is just one, ... but could be "subdatasets", thus giving ability to act on a pair of `{ds}` and `{subds}` for each `{ds}` to be traversed)?

Reply:
I will consider "parallelizing" across other entities out of scope for this PR -- not yet sure how it would be of help even, in particular at user API/interactions.